File: backend/justsupply/domain/supplier.py
from datetime import datetime
from enum import StrEnum
from uuid import UUID

from pydantic import BaseModel, ConfigDict, HttpUrl


class Commodity(StrEnum):
    COCOA = "cocoa"
    COFFEE = "coffee"


class Supplier(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: UUID
    legal_name: str
    country_code: str
    website: HttpUrl | None
    commodities: tuple[Commodity, ...]
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/supplier.ts
export type Commodity = 'cocoa' | 'coffee'

export interface Supplier {
  id: string
  legal_name: string
  country_code: string
  website: string | null
  commodities: Commodity[]
  created_at: string
  updated_at: string
}

export interface SupplierInput {
  legal_name: string
  country_code: string
  website: string | null
  commodities: Commodity[]
}

export interface SupplierListResponse {
  items: Supplier[]
  total: number
}
